testStatsBasedRepartitionDataOnInsert flaky

```
[ERROR] io.trino.plugin.iceberg.TestIcebergOrcConnectorTest.testStatsBasedRepartitionDataOnInsert  Time elapsed: 1.576 s  <<< FAILURE!
java.lang.AssertionError: 
[Rows for query [SELECT count(*) FROM "repartition_SELECTDISTINCTorderkeycustkeyorderstatusFROMtpchtinyordersinsert_orderstatus_1ns2vmr9lb$files"]] 
Expecting:
  <(3)>
to contain exactly in any order:
  <[(9)]>
elements not found:
  <(9)>
and elements not expected:
  <(3)>

	at io.trino.sql.query.QueryAssertions$QueryAssert.lambda$matches$2(QueryAssertions.java:346)
	at io.trino.sql.query.QueryAssertions$QueryAssert.matches(QueryAssertions.java:333)
	at io.trino.sql.query.QueryAssertions$QueryAssert.matches(QueryAssertions.java:328)
	at io.trino.plugin.iceberg.BaseIcebergConnectorTest.testRepartitionData(BaseIcebergConnectorTest.java:2874)
	at io.trino.plugin.iceberg.BaseIcebergConnectorTest.lambda$testStatsBasedRepartitionData$27(BaseIcebergConnectorTest.java:2831)
	at io.trino.testing.assertions.Assert.assertEventually(Assert.java:74)
	at io.trino.testing.assertions.Assert.assertEventually(Assert.java:66)
	at io.trino.testing.assertions.Assert.assertEventually(Assert.java:61)
	at io.trino.plugin.iceberg.BaseIcebergConnectorTest.testStatsBasedRepartitionData(BaseIcebergConnectorTest.java:2830)
	at io.trino.plugin.iceberg.BaseIcebergConnectorTest.testStatsBasedRepartitionDataOnInsert(BaseIcebergConnectorTest.java:2809)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

```

https://github.com/trinodb/trino/runs/5280874831?check_suite_focus=true

--- a/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java
+++ b/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java
@@ -17,6 +17,7 @@ import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.ImmutableSet;
 import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
 import io.trino.Session;
 import io.trino.hdfs.HdfsContext;
 import io.trino.metadata.Metadata;
@@ -122,6 +123,7 @@ import static java.time.format.DateTimeFormatter.ISO_OFFSET_DATE_TIME;
 import static java.util.Collections.nCopies;
 import static java.util.Objects.requireNonNull;
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.MINUTES;
 import static java.util.stream.Collectors.joining;
 import static java.util.stream.Collectors.toList;
 import static java.util.stream.Collectors.toUnmodifiableList;
@@ -4191,7 +4193,7 @@ public abstract class BaseIcebergConnectorTest
                 3);
         // Test uses relatively small table (60K rows). When engine doesn't redistribute data for writes,
         // occasionally a worker node doesn't get any data and fewer files get created.
-        assertEventually(() -> {
+        assertEventually(new Duration(3, MINUTES), () -> {
             testRepartitionData(
                     sessionRepartitionMany,
                     sourceRelation,

